### PR TITLE
Improve function invocation error

### PR DESCRIPTION
**Pull request by elsteelbrain**

### CODE REVIEW

File: function/function.go [hunk 1]
```diff
@@ -137,6 +137,10 @@ func (f *Function) callAWSLambda(payload []byte) ([]byte, error) {
 	if err != nil {
 		if awserr, ok := err.(awserr.Error); ok {
 			switch awserr.Code() {
+			case "AccessDeniedException":
```
Comment (mthenw) [inline function/function.go]: @elsteelbrain I did it differently, please take a look

File: function/function.go [hunk 2]
```diff
@@ -128,23 +128,23 @@ func (f *Function) callAWSLambda(payload []byte) ([]byte, error) {
 		config = config.WithCredentials(credentials.NewStaticCredentials(f.Provider.AWSAccessKeyID, f.Provider.AWSSecretAccessKey, f.Provider.AWSSessionToken))
 	}
 
-	awsSession, err := session.NewSession(config)
-	if err != nil {
-		return nil, &ErrFunctionError{err}
-	}
-	awslambda := lambda.New(awsSession)
+	awslambda := lambda.New(session.New(config))
```
Comment (elsteelbrain) [inline function/function.go]: This is deprecated!

File: function/function.go (hunk 1 above)
Comment (elsteelbrain) [reply function/function.go]: Have you tested it?

File: function/function.go (hunk 2 above)
Comment (mthenw) [reply function/function.go]: ok, I will revert it

File: function/function.go [hunk 3]
```diff
@@ -128,23 +128,23 @@ func (f *Function) callAWSLambda(payload []byte) ([]byte, error) {
...
 
 	invokeOutput, err := awslambda.Invoke(&lambda.InvokeInput{
 		FunctionName: &f.Provider.ARN,
 		Payload:      payload,
 	})
 	if err != nil {
-		if receivedAWSErr, ok := err.(awserr.Error); ok {
-			switch receivedAWSErr.Code() {
+		if awserr, ok := err.(awserr.Error); ok {
```
Comment (elsteelbrain) [inline function/function.go]: `awserr` collides with an imported package

File: function/function.go (hunk 1 above)
Comment (mthenw) [reply function/function.go]: Yes, of course.

File: function/function.go (hunk 3 above)
Comment (mthenw) [reply function/function.go]: If there is no automated check for that (with `gometalinter`) I would assume it's fine.

Comment (elsteelbrain) [reply function/function.go]: My linter is reporting a warning on this. Not that it would be problematic, just that it is bad practice.

Comment (elsteelbrain) [reply function/function.go]: 👍

**Pull request merged**